src/net/socket.rs

```diff
@@ -0,0 +1,222 @@
+                            match self.recv_from() {
+                                Ok(Some(packet)) => {
+                                    if let Err(err) =
+                                        self.event_sender.send(SocketEvent::Packet(packet))
+                                    {
+                                        error!("{:?}", err);
+                                    }
+                                }
+                                Ok(None) => continue,
+                                Err(ref err) => match err.kind() {
+                                    NetworkErrorKind::IOError(io_err)
+                                        if io_err.kind() == io::ErrorKind::WouldBlock =>
+                                    {
+                                        break;
+                                    }
+                                    _ => error!("{:?}", err),
+                                },
+                            };
+                        }
+                    }
+                }
+                _ => unreachable!(),
+            }
+        }
+        Ok(())
+    }
+
+    /// Serializes and sends a `Packet` on the socket. On success, returns the number of bytes written.
+    fn send_to(&mut self, packet: Packet) -> NetworkResult<usize> {
+        let connection = self
+            .connections
+            .get_or_insert_connection(&packet.addr(), self.config.clone());
+        let mut packet_data =
+            connection.process_outgoing(packet.payload(), packet.delivery_method())?;
+        let mut bytes_sent = 0;
+
+        if let Some(link_conditioner) = &self.link_conditioner {
+            if link_conditioner.should_send() {
+                for payload in packet_data.parts() {
+                    bytes_sent += self.send_packet(&packet.addr(), &payload)?;
+                }
+            }
+        } else {
+            for payload in connection.gather_dropped_packets() {
+                bytes_sent += self.send_packet(&packet.addr(), &payload)?;
+            }
+
+            for payload in packet_data.parts() {
+                bytes_sent += self.send_packet(&packet.addr(), &payload)?;
+            }
+        }
+
+        Ok(bytes_sent)
+    }
+
+    /// Receives a single message from the socket. On success, returns the packet containing origin and data.
+    fn recv_from(&mut self) -> NetworkResult<Option<Packet>> {
+        let (recv_len, address) = self.socket.recv_from(&mut self.recv_buffer)?;
+        if recv_len == 0 {
+            return Err(NetworkErrorKind::ReceivedDataToShort)?;
+        }
+
+        let received_payload = &self.recv_buffer[..recv_len];
+        let connection = self
+            .connections
+            .get_or_insert_connection(&address, self.config.clone());
+        connection.process_incoming(received_payload)
+    }
+
+    /// Send a single packet over the udp socket.
+    fn send_packet(&self, addr: &SocketAddr, payload: &[u8]) -> NetworkResult<usize> {
```

Comment:
I prefer to have the same method name as the function what it is wrapping: `send_to`

Reply:
I'm inclined to leave this.